lib/rules/interface-name-prefix.js

```diff
@@ -43,19 +43,8 @@ module.exports = {
             if (typeof name !== "string") {
                 return false;
             }
-            if (name.length === 0) {
-                return false;
-            }
-            const first = name.charAt(0);
-            const second = name.charAt(1);
 
-            if (second === "") {
-                return false;
-            }
-            if (first !== "I" || second !== second.toUpperCase()) {
-                return false;
-            }
-            return true;
+            return !!name.match(/^I[A-Z]/);
```

Comment:
I like this solution! Could also use [`RegExp.prototype.test()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp/test), which returns a boolean, instead of having to cast the output of `match()` to a boolean:

```js
/^I[A-Z]/.test(name)
```

Reply:
I prefer that!
I always forget regex's have methods..